virttest/utils_test/qemu/__init__.py

```diff
@@ -186,6 +186,38 @@ def clear_win_driver_verifier(driver, vm, timeout=300):
         session.close()
 
 
+@error_context.context_aware
+def check_win_driver(test, driver, vm, timeout=300):
+    """
+    Check if driver is installed and state is running for windows guest.
+    :param test: QEMU test object.
+    :param driver: The driver which needs enable the verifier.
+    :param vm: VM object.
+    :param timeout: Timeout in seconds.
+    """
+    session = vm.wait_for_login(timeout=timeout)
+    logging.info("Sleep a while before check guest msinfo.")
+    time.sleep(10)
+    try:
+        error_context.context("Check %s driver state." % driver,
+                              logging.info)
+        msinfo_cmd = "start /wait msinfo32 /report c:\msinfo.txt"
+        driver_check_cmd = "type c:\msinfo.txt |findstr /i \"%s.sys\"" % driver
+        if session.cmd_status(msinfo_cmd, timeout=360):
+            test.error("Get msinfo32 failed.")
```

Comment:
This part maybe better to `return False` as well.
In fact, all these if clauses may better `return (bool , output)` , so you can check the bool, and test.error(output) accordingly in tp-qemu.
You can also remove these logging.info() there.

Reply:
I have modified these code, pls have another check.
Thanks.